docs/design/consistency.md

```diff
@@ -34,53 +34,53 @@ What specific guarantees does TigerBeetle provide to applications?
 - A client session never observes a broken invariant (e.g.
   [`flags.credits_must_not_exceed_debits`](../reference/accounts.md#flagscredits_must_not_exceed_debits)
   or [`flags.linked`](../reference/transfers.md#flagslinked)).
-- Multiple client sessions may receive replies [out of order](#reply-order) relative to one another.
+- Multiple client sessions may receive replies [out of order](#reply-order) relative to one another. For example, if two clients submit requests around the same time, the client whose request is commmitted first might receive the reply later.
 - A client session can consider a request executed when it receives a reply for the request.
 
-#### Requests
+#### [Requests](./client-requests.md)
+> Note that a request refers to a batch, rather than a single operation.
 
 - A request executes within the cluster at most once.
-- Requests do not [time out](#retries) — a timeout typically implies failure, which cannot be
-  conclusively determined in the context of network faults.
+- Requests do not [time out](#retries) (a timeout typically implies failure, which cannot be
```

Comment:
I think I don't totally understand what this means:
> A timeout typically implies failure, which cannot be conclusively determined in the context of network faults.

Reply:
Consider case A:

1. Client connects to cluster.
2. Client sends request X.
3. (There is a network partition between the client and the cluster -- the cluster never receives request X.)

And case B:

1. Client connects to cluster.
2. Client sends request X.
3. Cluster receives request X.
4. Cluster sends reply X to client.
5. (There is a network partition between the client and the cluster -- the client never receives reply X.)

From the client's point of view, these are indistinguishable!

If it were somehow possible to distinguish cases A and B at the client, then it would be safe for `client.request()` to throw a "not connected" error after some timeout in case A. But if the client threw a "not connected" error in case B, 
then the client/application would assume that X was not executed, which is incorrect!

As such, we handle these cases the same way -- the client keeps retrying a request until it receives a reply, at which point is is certain that the cluster has processed it.

The (big) downside here is that network (and configuration) issues exist -- right now we hide all issues, which can make troubleshooting difficult. In the future I expect we'll expose out-of-band warnings if we _suspect_ that the cluster is inaccessible.